root/user/UserProfile.js

```diff
@@ -0,0 +1,537 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import capitalize from 'lodash/capitalize';
+import * as React from 'react';
+
+import UserAccountLayout from '../components/UserAccountLayout';
+import {withCatalystContext} from '../context';
+import DescriptiveLink
+  from '../static/scripts/common/components/DescriptiveLink';
+import commaList from '../static/scripts/common/i18n/commaList';
+import expand2react from '../static/scripts/common/i18n/expand2react';
+import bracketed, {bracketedText}
+  from '../static/scripts/common/utility/bracketed';
+import nonEmpty from '../static/scripts/common/utility/nonEmpty';
+import {formatCount, formatPercentage} from '../statistics/utilities';
+import formatUserDate from '../utility/formatUserDate';
+import {canNominate} from '../utility/voting';
+
+function generateUserTypesList(user: EditorT) {
+  const typesList = [];
+  if (user.deleted) {
+    typesList.push(l('Deleted User'));
+  }
+  if (user.is_auto_editor) {
+    typesList.push(exp.l('{doc|Auto-Editor}', {doc: '/doc/Editor'}));
```

Comment:
Bonus: While you are at it, would it be ok to link to the specific anchor for every link to this doc page?